### PR TITLE
MCLOUD-2789: Add custom ES plugins

### PR DESCRIPTION
**Pull request by oshmyheliuk**
### Description
https://jira.corp.magento.com/browse/MCLOUD-2789
Added functionality for installing custom elastic search plugins.

### Manual testing scenarios

https://studio.cucumber.io/projects/210147/test-plan/folders/1564670/scenarios/5121522

1. Build elasticsearch image:
```
docker build -t test/es:latest ../magento-cloud-docker/images/elasticsearch/7.7
```
2. Create `docker-compose.override.yaml` file
```
version: "2.1"

services:
    elasticsearch:
        image: 'test/es:latest'
```
3. Add some custom plugins in `.magento/services.yaml`, for example:
```
elasticsearch:
    type: elasticsearch:7.7
    disk: 1024
    configuration:
        plugins:
            - analysis-icu
            - analysis-phonetic
            - analysis-stempel
            - analysis-nori
```
4. Run command for generating docker-compose.yaml
```
vendor/bin/ece-docker build:compose
```
5. Open `docker-compose.yaml` file and check that elasticsearch service contains:
```
    environment:
      - 'ES_PLUGINS=analysis-stempel analysis-nori'
```
P.S. `analysis-icu` and `analysis-phonetic` are skipped as they installed by default.

6. Run `docker-compose up -d`
7. Run `docker-compose logs elasticsearch` and check that added plugin was installed.
```
elasticsearch_1  | Intalling plugins:
elasticsearch_1  | -> Installing analysis-stempel
elasticsearch_1  | -> Downloading analysis-stempel from elastic
elasticsearch_1  | -> Installed analysis-stempel
elasticsearch_1  | -> Installing analysis-nori
elasticsearch_1  | -> Downloading analysis-nori from elastic
elasticsearch_1  | -> Installed analysis-nori
```
8. Repeat with different elasticsearch version >= 6.5
### Release notes

Added support to use the Elasticsearch plugin configuration information from the `services.yaml` file to generate the `docker-compose.yaml` file for a Cloud Docker environment.

### Associated documentation updates

https://github.com/magento/devdocs/pull/7898

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] Pull request introduces user-facing changes and includes meaningful release notes and documentation
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

File: src/Command/Image/GenerateEs.php [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    private const SINGLE_NODE = 'RUN echo "discovery.type: single-node" >> '
+        . '/usr/share/elasticsearch/config/elasticsearch.yml';
+
+    /**
+     * Configuration map for generating es images data
+     *
+     * @var array
+     */
+    private $versionMap = [
+        '6.5' => [
```
Comment (NadiyaS) [inline src/Command/Image/GenerateEs.php:32]: don't we support more earlier versions as we do on cloud?

Comment (oshmyheliuk) [reply src/Command/Image/GenerateEs.php:32]: They already reached EOL, I don't think we still need to add new features to old deprecated versions.

**Review by arhiopterecs — APPROVED** [review]

**Review by NadiyaS — APPROVED** [review]

**Comment by mveeramneni** [comment]
QA Approved

**Comment by meker12** [comment]
@oshmyheliuk  Remember to add the release notes and documentation PR to the PR description when applicable.  Thanks!

**Pull request merged**